doc/development/submitting/general.md

```diff
@@ -0,0 +1,353 @@
+SVN branches.
+
+### Comments
+
+PLEASE take the time to add comments throughout your code explaining what the
+code is doing. It will save a HUGE amount of time and frustration for other
+programmers that may have to change your code in the future.
+
+### End Of Line
+
+Make sure a new line is at the end of each file and UNIX style newlines are
+used (`\n`).
+
+### Branches and backports
+
+_This section applies to the core developers with access to the main repository
+(other contributors can safely ignore it)._
+
+GRASS GIS Subversion repository has trunk and several branches. All the development
+should happen in trunk (trunk is a "development branch"). All the other branches
+are usually associated with releases and should contain stable code which is being
+prepared towards the given release.
+
+When a bug is fixed the fix should be committed to trunk, tested there, and than
+backported to the release branch or branches. The testing before backport should
+include user testing, running automated test (if available), and compilation of
+the whole source tree (ideally after `make distclean`). Note that thinks like
+testing should be done also before the original commit to trunk. Also note that
+not all these steps has to be done manually, you can take an advantage of
+[​Travis CI](https://travis-ci.org/GRASS-GIS/) or
+[​automated runs of the GRASS GIS testing suite](http://fatra.cnr.ncsu.edu/grassgistests/).
+
+Often there is more than one active stable branch, if you are backporting, make
+sure you always backport to all the branches between trunk and the furthest
+branch you are backporting to. For example, let's say we have trunk and branches
+7.2 and 7.0, if you backport to 7.0, you have to also backport to 7.2. You can
+also choose to backport only to the closest branch, in our example 7.2.
+
+Backport only complete fixes. When you are not sure if the fix is complete or if
+there is an possibility that some details such as wording will change, wait with
+the backport for a little while and backport all the changes together to reduce
+number of commits which needs to be reviewed (right now or in the future). You
+can also backport multiple commits from trunk as one commit if you think it is
+appropriate.
+
+Include the number of the commit (or commits) you are backporting, into the commit
+message, for example: `less verbose messages (backport r89436)`. This will help
+matching the file content in between branches and tracking if the commits were
+backported.
+
+As a developer you should maintain a list of commits which you plan to backport.
+One way how to do it is to subscribe to the
+[​grass-commit mailing list](https://lists.osgeo.org/listinfo/grass-commit) and
+filter your commits in your email.
+
+Do not add svn merge property to the commit. Review
+[tools directory in Addons repository](/grass/browser/grass-addons/tools) for
+scripts which will help you with backporting.
```

Comment:
```suggestion
The GRASS GIS repository on GitHub has the "main" and several release branches. 
All the development should happen in "main" (the development branch). All the other
branches are usually associated with releases and should contain stable code
which is being prepared towards the given release.

When a bug is fixed the fix should be committed to "main", tested there, and then
backported to the release branch or branches, if appropriate. The testing before
doing a backport should include user testing, running of automated test (if
available), and compilation of the whole source tree (ideally after the
`make distclean` step). Note that testing should already be done also prior to
the original commit to "main". Also note that not all these steps have to be
done manually, you can take an advantage of our [​CI](https://github.com/OSGeo/grass/actions) or
[​automated runs of the GRASS GIS testing suite](http://fatra.cnr.ncsu.edu/grassgistests/).

Often there is more than one active release branch, if you are backporting, make
sure you always backport to all the branches between "main" and the furthest
branch you are backporting to. For example, let's say we have "main" and branches
8.2 and 7.8, if you backport to 8.2, you should also consider to backport to 7.8.
You can also choose to backport only to the closest branch, in our example 8.2.

Backport only complete fixes. When you are not sure if the fix is complete or if
there is an possibility that some details such as wording will change, wait with
the backport for a little while and backport all the changes together to reduce
number of commits which needs to be reviewed (right now or in the future). You
can also backport multiple commits from "main" as one commit if you think it is
appropriate.

Include the number of the pull request when you are backporting, into the commit
message, for example: `less verbose messages (backport of PR #1234)`. This will help
matching the file content in between branches and tracking if the commits were
backported.
```

Attempt to update this section